common/autoware_universe_utils/include/autoware/universe_utils/ros/managed_transform_buffer.hpp

```diff
@@ -124,42 +189,38 @@ class StaticTransformBuffer
     return true;
   }
 
-  /**
-   * Transforms a point cloud from one frame to another.
+  /** @brief Retrieves a transform between two dynamic frames.
    *
-   * @param node A pointer to the ROS node.
-   * @param target_frame The target frame to transform the point cloud to.
-   * @param cloud_in The input point cloud to be transformed.
-   * @param cloud_out The transformed point cloud.
-   * @return True if the transformation is successful, false otherwise.
+   * This function attempts to retrieve a transform between the target frame and the source frame.
+   * If success, the transform matrix is set to the output parameter and the function returns true.
+   * Otherwise, transform matrix is set to identity and the function returns false.
+   *
+   * @param target_frame The target frame.
+   * @param source_frame The source frame.
+   * @param eigen_transform The output Eigen transform matrix. Is set to identity if the transform
+   * is not found.
+   * @return True if the transform was successfully retrieved, false otherwise.
    */
-  bool transformPointcloud(
-    rclcpp::Node * node, const std::string & target_frame,
-    const sensor_msgs::msg::PointCloud2 & cloud_in, sensor_msgs::msg::PointCloud2 & cloud_out)
+  bool getDynamicTransform(
+    const std::string & target_frame, const std::string & source_frame,
+    Eigen::Matrix4f & eigen_transform)
   {
-    if (
-      pcl::getFieldIndex(cloud_in, "x") == -1 || pcl::getFieldIndex(cloud_in, "y") == -1 ||
-      pcl::getFieldIndex(cloud_in, "z") == -1) {
-      RCLCPP_ERROR(node->get_logger(), "Input pointcloud does not have xyz fields");
-      return false;
-    }
-    if (target_frame == cloud_in.header.frame_id) {
-      cloud_out = cloud_in;
-      return true;
-    }
-    Eigen::Matrix4f eigen_transform;
-    if (!getTransform(node, target_frame, cloud_in.header.frame_id, eigen_transform)) {
+    auto tf = tf_listener_->getTransform(
+      target_frame, source_frame, rclcpp::Time(0), rclcpp::Duration(1000ms));
+    if (tf == nullptr) {
+      eigen_transform = Eigen::Matrix4f::Identity();
       return false;
     }
-    pcl_ros::transformPointCloud(eigen_transform, cloud_in, cloud_out);
-    cloud_out.header.frame_id = target_frame;
+    pcl_ros::transformAsMatrix(*tf, eigen_transform);
     return true;
   }
 
-private:
   TFMap buffer_;
+  rclcpp::Node * node_;
```

Comment:
can you const this or use a reference?

Reply:
Addressed in https://github.com/autowarefoundation/autoware.universe/pull/8678/commits/d905dfe990ddd1186289d5f541a9d7eacafa666f